### PR TITLE
New: Add scan_dependencies and scan_secrets

### PR DESCRIPTION
**Pull request by ewokcami**
Add:

1. `scan_dependencies`
2. `scan_secrets`
3. `scan_git_logs`

\+ Update README accordingly

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -136,6 +140,8 @@ jobs:
         uses: endorlabs/github-action@main
         with:
           namespace: "example"
+          scan_dependencies: "true"
+          scan_secrets: "true"
```
Comment (dkourkouzelis) [inline README.md:143]: Do we want to enable this by default? Prior to this PR, the github action would not enable secret scanning, right?

Comment (IAmATeaPot418) [reply README.md:143]: I think its fair to do secrets by default as long as we aren't doing the full history scan by default.

**Comment by IAmATeaPot418** [comment]
Do we need pre_commit_checks here? Given that the GitHub action is going to scan post commit always is there a use case for this? @ewokcami

File: README.md (hunk 1 above)
Comment (ewokcami) [reply README.md:143]: Right now only dependencies (git,analytics) are enabled by default

Comment (IAmATeaPot418) [reply README.md:143]: Stick with consistency

**Review by IAmATeaPot418 — APPROVED** [review]

**Pull request merged**